overlord/assertstate/validation_set_tracking.go

```diff
@@ -0,0 +1,138 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package assertstate
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/snapcore/snapd/overlord/state"
+)
+
+// ValidationMode reflects the mode of respective validation set, which is
+// either monitoring or enforcing.
+type ValidationMode int
+
+const (
+	Monitor ValidationMode = iota
+	Enforce
+)
+
+// ValidationTrackingKey is a key for tracking the given validation set.
+type ValidationTrackingKey struct {
+	AccoundID string
+	Name      string
+}
+
+func (v *ValidationTrackingKey) String() string {
+	return fmt.Sprintf("%s/%s", v.AccoundID, v.Name)
+}
+
+// MarshalText implements TextMarshaler and allows for using ValidationTrackingKey
+// as map key when marshaling into JSON.
+func (v ValidationTrackingKey) MarshalText() (text []byte, err error) {
+	return []byte(v.String()), nil
+}
+
+// UnmarshalText implements TextUnmarshaler and allows for using ValidationTrackingKey
+// as map key when unmarshaling from JSON.
+func (v *ValidationTrackingKey) UnmarshalText(text []byte) error {
+	parts := strings.Split(string(text), "/")
+	if len(parts) != 2 {
+		return fmt.Errorf("invalid validation set tracking key: %q", string(text))
+	}
+	v.AccoundID = parts[0]
+	v.Name = parts[1]
+	return nil
+}
+
+// ValidationSetTracking holds tracking parameters for associated validation set.
+type ValidationSetTracking struct {
+	Mode ValidationMode `json:"mode"`
+
+	// PinnedSeq is an optional pinned sequence point, or 0 if not pinned.
+	PinnedSeq int `json:"pinned-seq,omitempty"`
+
+	// LastSeq is the last known sequence point obtained from the store.
```

Comment:
this is not going to be the last known one, but the one we have in use because it doesn't conflict, so probably "current" not last. We might need latest as well, but we'll need to see

Reply:
Done.